deltachat-ffi/deltachat.h

```diff
@@ -1178,6 +1178,68 @@ int dc_send_webxdc_status_update (dc_context_t* context, uint32_t msg_id, const
  */
 char* dc_get_webxdc_status_updates (dc_context_t* context, uint32_t msg_id, uint32_t serial);
 
+
+/**
+ * Set Webxdc file as integration.
+ * see dc_init_webxdc_integration() for more details about Webxdc integrations.
+ *
+ * @warning This is an experimental API which may change in the future
+ *
+ * @memberof dc_context_t
+ * @param context The context object.
+ * @param file The .xdc file to use as Webxdc integration.
+ */
+void             dc_set_webxdc_integration (dc_context_t* context, const char* file);
```

Comment:
how would the core know the type of integration? from a property in the manifest.toml file? (maybe it is better to have some temporary on-focus api specifically for maps integration and then in the future properly define the generic API)

Reply:
we had a `type` parameter there before, but that turns out to be too confusing.

might be an additional parameter is superfluous anyways as the type of the integration may be determined by some manifest entry in the future. but let's see, nothing to worry about now

Reply:
> This API could have been named `dc_set_webxdc_map`.

yes, and worse is `dc_init_webxdc_integration` there is no control over which integration is being initialized for the given chat so it is confusing, if it is only for map it would be more clear if it states so, since if it would support other integrations in the future, it would need to change anyway to accept some parameter with the integration "id" or so :thinking:

Reply:
(but this is just some soft-opinion/comemnts, I don't want to block this PR, and since the use is internal for now it doesn't really matter so don't mind me ;)

Reply:
yes, `dc_init_webxdc_integration` would need a `type` if we would support other integrations. the "integrate for" is maybe easier to read as "data source" or so, idk